soc/xtensa/esp32/Kconfig.soc

```diff
@@ -7,6 +7,10 @@ config SOC_ESP32
 	select CLOCK_CONTROL
 	select CLOCK_CONTROL_ESP32
 	select DYNAMIC_INTERRUPTS
+	select HAS_FLASH_LOAD_OFFSET
+	select BOOTLOADER_MCUBOOT
+	select MCUBOOT_GENERATE_UNSIGNED_IMAGE
```

Comment:
I don't think the SoC in use should decide the bootloader to use.
Seems like the wrong location for such knowledge to me.

Secondly, the SoCs in Zephyr might be used in other places as well, like when building MCUBoot for Zephyr which will result in:
```
warning: BOOTLOADER_MCUBOOT (defined at Kconfig.zephyr:554) has direct dependencies !MCUBOOT with value n, but is currently being y-selected by the following symbols:
 - SOC_ESP32 (defined at soc/xtensa/esp32/Kconfig.soc:4), with value y, direct dependencies <choice> (value: y), and select condition <choice> (value: y)

error: Aborting due to Kconfig warnings
```
```

Reply:
I updated this to only enable this symbols if `BOOTLOADER_MCUBOOT` is set in SoC level. PTAL.